pkg/loop/internal/relayer/pluginprovider/ext/ccip/test/commit_store_test.go

```diff
@@ -41,63 +34,25 @@ func TestStaticCommitStore(t *testing.T) {
 
 func TestCommitStoreGRPC(t *testing.T) {
 	t.Parallel()
-	ctx := tests.Context(t)
-	// create a price registry server
-	port := freeport.GetOne(t)
-	addr := fmt.Sprintf("localhost:%d", port)
-	lis, err := net.Listen("tcp", addr)
-	require.NoError(t, err, "failed to listen on port %d", port)
-	t.Cleanup(func() { lis.Close() })
-	// we explicitly stop the server later, do not add a cleanup function here
-	testServer := grpc.NewServer()
-	// handle client close and server stop
-	shutdown := make(chan struct{})
-	closer := &serviceCloser{closeFn: func() error { close(shutdown); return nil }}
-	lggr := logger.Test(t)
-	broker := &loopnettest.Broker{T: t}
-	brokerExt := &loopnet.BrokerExt{
-		Broker:       broker,
-		BrokerConfig: loopnet.BrokerConfig{Logger: lggr, StopCh: make(chan struct{})},
-	}
-	offRamp, err := ccip.NewCommitStoreReaderGRPCServer(CommitStoreReader, brokerExt)
-	require.NoError(t, err)
-	offRamp = offRamp.AddDep(closer)
-
-	ccippb.RegisterCommitStoreReaderServer(testServer, offRamp)
-	// start the server and shutdown handler
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		require.NoError(t, testServer.Serve(lis))
-	}()
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		<-shutdown
-		t.Log("shutting down server")
-		testServer.Stop()
-	}()
-	// create a price registry client
-	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	require.NoError(t, err, "failed to dial %s", addr)
-	t.Cleanup(func() { conn.Close() })
-	client := ccip.NewCommitStoreReaderGRPCClient(conn, brokerExt)
-
-	// test the client
-	roundTripCommitStoreTests(ctx, t, client)
-	// closing the client executes the shutdown callback
-	// which stops the server.  the wg.Wait() below ensures
-	// that the server has stopped, which is what we care about.
-	cerr := client.Close()
-	require.NoError(t, cerr, "failed to close client %T, %v", cerr, status.Code(cerr))
-	wg.Wait()
+
+	scaffold := looptest.NewGRPCScaffold(t, setupCommitStoreServer, ccip.NewCommitStoreReaderGRPCClient)
+	roundTripCommitStoreTests(t, scaffold.Client())
+	// commit store implements dependency management, test that it closes properly
+	t.Run("Dependency management", func(t *testing.T) {
+		d := &looptest.MockDep{}
+		scaffold.Server().AddDep(d)
+		assert.False(t, d.IsClosed())
+		scaffold.Client().Close()
+		assert.True(t, d.IsClosed())
+	})
 }
 
 // roundTripCommitStoreTests tests the round trip of the client<->server.
 // it should exercise all the methods of the client.
 // do not add client.Close to this test, test that from the driver test
-func roundTripCommitStoreTests(ctx context.Context, t *testing.T, client cciptypes.CommitStoreReader) {
+func roundTripCommitStoreTests(t *testing.T, client cciptypes.CommitStoreReader) {
+	ctx := tests.Context(t)
```

Comment:
It would be safer to make a context within each sub-test

Reply:
why? the are all the same `t *testing.T` 

if we make a ctx for each subtest, wouldn't every single one of them behave in the exact same way?

Reply:
Every sub-test derives a new `t *testing.T`:
```go
t.Run("", func(t *testing.T) {
```